api/app/routers/pteams.py

```diff
@@ -157,6 +157,28 @@ def get_pteam(
     return _extend_pteam_tags(pteam)
 
 
+@router.get("/{pteam_id}/groups", response_model=schemas.PTeamGroupResponse)
+def get_pteam_groups(
+    pteam_id: UUID,
+    current_user: models.Account = Depends(get_current_user),
+    db: Session = Depends(get_db),
+):
+    """
+    Get groups of the pteam.
+    """
+    pteam = validate_pteam(db, pteam_id, on_error=status.HTTP_404_NOT_FOUND)
+    assert pteam
+    check_pteam_membership(db, pteam_id, current_user.user_id, on_error=status.HTTP_403_FORBIDDEN)
+    set_groups = set()
+    for pteamtag in pteam.pteamtags:
```

Comment:
pteamtags はリレーションなので、N+1問題が発生しそうです (pteamtags の各レコードごとにselect 分を発行してしまう問題)

joined loadを使うのが良さそうです
https://docs.sqlalchemy.org/en/20/orm/queryguide/relationships.html#sqlalchemy.orm.joinedload

```suggestion
    pteam = db.query(models.Pteam).options(joinedload(models.Pteam.pteamtags)).get(pteam_id)
    for pteamtag in pteam.pteamtags:
```